src/main/webapp/app/editor/ace/editor-ace.component.ts

```diff
@@ -0,0 +1,199 @@
+
+            if (!this.editorFileSessions[fileName]) {
+                this.editorFileSessions[fileName] = {};
+                this.editorFileSessions[fileName].code = fileObj.fileContent;
+            }
+            /**
+             * Assign the obtained file content to the editor and set the ace mode
+             * Additionally, we resize the editor window and set focus to it
+             */
+            this.editorText = fileObj.fileContent;
+            // This fetches a list of all supported editor modes and matches it afterwards against the file extension
+            const aceModeList = ace.require('ace/ext/modelist');
+            const aceMode = aceModeList.getModeForPath(fileName);
+            this.editor.setMode(aceMode);
+            this.editor.getEditor().resize();
+            this.editor._editor.focus();
+        }, err => {
+            console.log('There was an error while getting file', this.fileName, err);
+        });
+    }
+
+    /**
+     * @function saveFile
+     * @desc Saved the currently selected file; is being called when the file is changed (onFileChanged)
+     * @param fileName: name of currently selected file
+     */
+    saveFile(fileName: string) {
+        if (this.onSaveStatusChange) {
+            this.onSaveStatusChange({
+                isSaved: false,
+                saveStatusLabel: ' <i class="fa fa-circle-o-notch fa-spin text-info"></i><span class="text-info"> Saving file.</span>'
+            });
+        }
+
+        this.repositoryFileService.update(this.participation.id,
+            fileName,
+            this.editorFileSessions[fileName].code)
+            .debounceTime(3000)
+            .distinctUntilChanged()
+            .subscribe(() => {
+                this.editorFileSessions[fileName].unsavedChanges = false;
+                this.updateSaveStatusLabel();
+        }, err => {
+            if (this.onSaveStatusChange) {
+                this.onSaveStatusChange( {
+                    isSaved: false,
+                    saveStatusLabel: '<i class="fa fa-times-circle text-danger"></i> <span class="text-danger"> Failed to save file.</span>'
+                });
+            }
+            console.log('There was an error while saving file', this.fileName, err);
+        });
+    }
+
+    /**
+     * @function onFileTextChanged
+     * @desc Callback function for text changes in the Ace Editor
+     * @param code {string} Current editor code
+     */
+    onFileTextChanged(code) {
+        console.log('onFileTextChanged', this.editorFileSessions, this.editorText, this.fileName);
+        if (this.editorFileSessions[this.fileName] !== code && this.editorText !== '') {
+            this.editorFileSessions[this.fileName].code = code;
+            this.editorFileSessions[this.fileName].unsavedChanges = true;
+
+            // Delay saving file by 2 seconds
+            setTimeout(() => {
+                this.saveFile(this.fileName);
+            }, 2500);
```

Comment:
please store 2500 as constant in the beginning of the file and give it a descriptive name, also make sure the comment is correct (2 vs 2.5 sec)